### ISSUE TITLE
mapContainersWithName: RUNTIME ERROR: Field does not exist: Inter

### ISSUE THREAD

**Issue by stjohnjohnson** [ksonnet/ksonnet-lib#120]
I'm trying to add a environments.  When doing so, I get a `RUNTIME ERROR` on k8s version 1.8 and not 1.7.

## Steps to Reproduce

```bash
$ ks init foo --api-spec=version:v1.7.0
$ cd foo && ks generate deployed-service demo --image=gcr.io/kuar-demo/kuard-amd64:1 --replicas=2 --type=ClusterIP
$ ks init bar --api-spec=version:v1.8.0
$ cd bar && ks generate deployed-service demo --image=gcr.io/kuar-demo/kuard-amd64:1 --replicas=2 --type=ClusterIP
$ ks version
ksonnet version: 0.9.0
jsonnet version: v0.9.5
client-go version: 1.8
```

Contents of `foo/components/demo.jsonnet` and `bar/components/demo.jsonnet`:

```jsonnet
local env = std.extVar("__ksonnet/environments");
local params = std.extVar("__ksonnet/params").components.demo;
local k = import "k.libsonnet";
local deployment = k.apps.v1beta1.deployment;
local container = k.apps.v1beta1.deployment.mixin.spec.template.spec.containersType;
local containerPort = container.portsType;
local service = k.core.v1.service;
local servicePort = k.core.v1.service.mixin.spec.portsType;
local envType = container.envType;

local targetPort = params.containerPort;
local labels = {app: params.name};

local appService = service
  .new(
    params.name,
    labels,
    servicePort.new(params.servicePort, targetPort))
  .withType(params.type);

local appDeployment = deployment
  .new(
    params.name,
    params.replicas,
    container
      .new(params.name, params.image)
      .withPorts(containerPort.new(targetPort)),
    labels) +
  deployment.mapContainersWithName(
    [params.name],
    function(c) c + container.withEnv([
      envType.new("FOO", "BAR"),
    ])
  );

k.core.v1.list.new([appService, appDeployment])
```

Execute the following command:
```bash
# foo dir
$ ks show default
---
apiVersion: v1
kind: Service
metadata:
  name: demo
spec:
  ports:
  - port: 80
    targetPort: 80
  selector:
    app: demo
  type: ClusterIP
---
apiVersion: apps/v1beta1
kind: Deployment
metadata:
  name: demo
spec:
  replicas: 2
  template:
    metadata:
      labels:
        app: demo
    spec:
      containers:
      - env:
        - name: FOO
          value: BAR
        image: gcr.io/kuar-demo/kuard-amd64:1
        name: demo
        ports:
        - containerPort: 80
# bar dir
$ ks show default
ERROR generate objects for namespace : unable to read /tmp/bar/environments/default/main.jsonnet: RUNTIME ERROR: Field does not exist: Inter
-------------------------------------------------
	/tmp/bar/lib/v1.8.0/k.libsonnet:15:40-49	thunk from <function <inNameSet>>

    local inNameSet(name) = std.length(std.Inter(nameSet, std.set([name]))) > 0;

-------------------------------------------------
	<builtin>	builtin function <length>

-------------------------------------------------
	/tmp/bar/lib/v1.8.0/k.libsonnet:15:29-80	function <inNameSet>

    local inNameSet(name) = std.length(std.Inter(nameSet, std.set([name]))) > 0;

-------------------------------------------------
	/tmp/bar/lib/v1.8.0/k.libsonnet:17:67-84	function <anonymous>

    self.mapContainers(function(c) if std.objectHas(c, "name") && inNameSet(c.name) then f(c) else c),

-------------------------------------------------
	/tmp/bar/lib/v1.8.0/k.libsonnet:17:39-84	function <anonymous>

    self.mapContainers(function(c) if std.objectHas(c, "name") && inNameSet(c.name) then f(c) else c),

-------------------------------------------------
	<std>:158:56-68	function <anonymous>

            std.makeArray(std.length(arr), function(i) func(arr[i])),

-------------------------------------------------
	During manifestation
```

## Remedy

This can be temporarily patched by changing that line to `std.Inter` to `std.setInter`

**Comment by bryanl** [comment]
This was fixed in ksonnet 0.9.1.

**Comment by stjohnjohnson** [comment]
Thanks! Confirmed.
